hackathon_site/review/tests.py

```diff
@@ -1,3 +1,356 @@
+
+        quantity_after = len(
+            Review.objects.filter(
+                decision_sent_date__isnull=True, status="Accepted"
+            ).all()
+        )
+
+        self.assertEqual(len(mail.outbox), 7)
+        self.assertRedirects(response, self.view)
+        self.assertEqual(quantity_before, quantity_after + 7)
+
+    def test_send_all_waitlisted_within_current_date_range(self):
+        self._login()
+        self._create_teams_and_reviews_for_mail_tests(date_offset=5)
+
+        quantity_before = len(
+            Review.objects.filter(
+                decision_sent_date__isnull=True, status="Waitlisted"
+            ).all()
+        )
+
+        # Send 10 waitlisted emails. There's only 2 waitlisted people, and 1 older. So should
+        # only send 1 emails
+        self.form_data["quantity"] = 10
+        self.form_data["status"] = "Waitlisted"
+
+        response = self.client.post(self.view, data=self.form_data)
+
+        quantity_after = len(
+            Review.objects.filter(
+                decision_sent_date__isnull=True, status="Waitlisted"
+            ).all()
+        )
+
+        self.assertEqual(len(mail.outbox), 1)
+        self.assertRedirects(response, self.view)
+        self.assertEqual(quantity_before, quantity_after + 1)
+
+    def test_send_all_rejected_within_current_date_range(self):
+        self._login()
+        self._create_teams_and_reviews_for_mail_tests(date_offset=5)
+
+        quantity_before = len(
+            Review.objects.filter(
+                decision_sent_date__isnull=True, status="Rejected"
+            ).all()
+        )
+
+        # Send 10 rejected emails. There's only 6 rejected people, and 2 older. So should
+        # only send 4 emails
+        self.form_data["quantity"] = 10
+        self.form_data["status"] = "Rejected"
+
+        response = self.client.post(self.view, data=self.form_data)
+
+        quantity_after = len(
+            Review.objects.filter(
+                decision_sent_date__isnull=True, status="Rejected"
+            ).all()
+        )
+
+        self.assertEqual(len(mail.outbox), 4)
+        self.assertRedirects(response, self.view)
+        self.assertEqual(quantity_before, quantity_after + 4)
+
+    def test_correct_text_in_accepted_email(self):
+        self._login()
+        self._create_teams_and_reviews_for_mail_tests()
+
+        # Send 1 accepted email
+        self.client.post(self.view, data=self.form_data)
+
+        clean = re.compile("<.*?>")
+        clean_mail_body = re.sub(clean, "", mail.outbox[0].body)
+
+        self.assertIn(
+            f"Congratulations, you’ve been accepted to { settings.HACKATHON_NAME }",
+            mail.outbox[0].subject,
+        )
+        self.assertIn(
+            f"The {settings.HACKATHON_NAME} team has reviewed your application, and we’re excited to welcome you to CoolHacks!",
+            clean_mail_body,
+        )
+
+    def test_correct_text_in_waitlisted_email(self):
```

Comment:
The waitlisted email currently renders the `final_review_response_date` wrong, it needs a `strptime` in there (it's just sticking the datetime in there in 2020-10-07 00:00:00-05:18 format right now).